dotnet/samples/Concepts/FunctionCalling/FunctionCalling.cs

```diff
@@ -0,0 +1,459 @@
+///    request to the AI model only and stops advertising them in subsequent requests to prevent an infinite loop where the model keeps calling functions repeatedly.
+///    The <see cref="FunctionChoiceBehavior.None"/> behavior tells the AI model to use the provided functions without calling them to generate a response.
+///    This behavior is useful for dry runs when you want to see which functions the model would call without actually invoking them.
+///
+/// ** Function invocation **
+///    The <see cref="FunctionChoiceBehavior.Auto"/> and <see cref="FunctionChoiceBehavior.Required"/> supports two modes of function invocation: manual and automatic:
+///    * Automatic function invocation mode enables invocation of all functions called by AI model automatically by SK.
+///      The results of these function invocations are added to the chat history and sent to the model automatically in the following request.
+///      The model then reasons about the chat history and then calls functions again or generates the final response.
+///      This approach is fully automated and requires no manual intervention from the caller. The automatic invocation mode is enabled by default.
+///    * Manual invocation mode returns all function calls requested by the AI model to the AI API caller. The caller is fully responsible
+///      for the invocation phase where they may decide which function to call, how to handle exceptions, call them in parallel or sequentially, etc.
+///      The caller then adds the function results/exceptions to the chat history and returns it to the model, which reasons about it
+///      and then calls functions again or generates the final response. This invocation mode provides more control over the function invocation phase to the caller.
+///      To enable manual invocation, the caller needs to set the `autoInvoke` parameter to `false` when specifying either <see cref="FunctionChoiceBehavior.Auto"/>
+///      or <see cref="FunctionChoiceBehavior.Required"/> in the <see cref="PromptExecutionSettings"/>.
+///
+///    SK supports only sequential invocation of functions in the automatic invocation mode at the moment. The concurrent invocation mode is only possible
+///    in the manual invocation mode.
+/// </summary>
+public class FunctionCalling(ITestOutputHelper output) : BaseTest(output)
+{
+    /// <summary>
+    /// This example demonstrates usage of <see cref="FunctionChoiceBehavior.Auto"/> that advertises all kernel functions to the AI model and invokes them automatically.
+    /// </summary>
+    [Fact]
+    public async Task RunPromptWithAutoFunctionChoiceBehaviorAdvertisingAllKernelFunctionsInvokedAutomaticallyAsync()
+    {
+        Kernel kernel = CreateKernel();
+
+        OpenAIPromptExecutionSettings settings = new() { FunctionChoiceBehavior = FunctionChoiceBehavior.Auto() };
```

Comment:
Out of scope of this PR: It would be nice to have some migration guide which will show previous and new approaches side-by-side.

Reply:
It will be added either in documentation or in the blogpost.